File: src/main/java/com/example/sneakerx/entities/ShopOrder.java
package com.example.sneakerx.entities;

import com.example.sneakerx.entities.enums.OrderStatus;
import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;
import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Entity
@Table(name = "shop_orders")
public class ShopOrder {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer shopOrderId;

    @ManyToOne
    @JoinColumn(name = "parent_order_id")
    @JsonIgnore // <--- THE FIX: Stops Jackson from serializing the Cart back to the parent
    @ToString.Exclude // <--- Stop Lombok from printing it
    @EqualsAndHashCode.Exclude
    private Order order;

    @ManyToOne
    @JoinColumn(name = "shop_id")
    @JsonIgnore // <--- THE FIX: Stops Jackson from serializing the Cart back to the parent
    @ToString.Exclude // <--- Stop Lombok from printing it
    @EqualsAndHashCode.Exclude
    private Shop shop;

    private Double shippingFee;
    private Double subTotal;

    @Enumerated(EnumType.STRING)
    private OrderStatus orderStatus;

    private String noteToSeller;

    @CreationTimestamp
    @Column(updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    private LocalDateTime updatedAt;

    @OneToMany(mappedBy = "shopOrder", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<OrderItem> orderItems;
}

File: lib/src/models/shop_order.dart
import 'package:sneakerx/src/models/enums/order_status.dart';
import 'package:sneakerx/src/models/order_item.dart';
import 'package:sneakerx/src/models/shops.dart';

class ShopOrder {
  final int shopOrderId;
  final int orderId;
  final Shop shop;
  final double shippingFee;
  final double subTotal;
  final OrderStatus orderStatus;
  final String noteToSeller;
  final DateTime createdAt;
  final List<OrderItem> orderItems;

  ShopOrder({
    required this.shopOrderId,
    required this.orderId,
    required this.shop,
    required this.shippingFee,
    required this.subTotal,
    required this.orderStatus,
    required this.noteToSeller,
    required this.createdAt,
    required this.orderItems,
  });
  
  factory ShopOrder.fromJson(Map<String, dynamic> json) {
    return ShopOrder(
      shopOrderId: json['shopOrderId'],
      orderId: json['orderId'],
      shop: Shop.fromJson(json['shop'] as Map<String, dynamic>),
      shippingFee: json['shippingFee'],
      subTotal: json['subTotal'],
      orderStatus: OrderStatus.values.byName(json['orderStatus']),
      noteToSeller: json['noteToSeller'],
      createdAt: DateTime.parse(json['createdAt']),
      orderItems: (json['orderItems'] as List?)
        ?.map((ele) => OrderItem.fromJson(ele as Map<String, dynamic>))
        .toList() ?? []
    );
  }
  
}